### DISCUSSION TITLE
Remove the paid editor sdk

### DISCUSSION THREAD

**Discussion by inson1** [FossifyOrg/General-Discussion#387] [Ideas]
_No description provided._

**Comment by Aga-C** [comment]
I'm closing this, because after quick research I found it's not true: https://www.gnu.org/licenses/gpl-faq.html.en#FSWithNFLibs. If you want to discuss using proprietary editor in the future, please use discussions.

EDIT: It was originally the reply to original quote from other discussion: "Per the GNU GPL, proprietary SDKs cannot be combined with GPL code in the same project. Either way, this proprietary library will have to be removed so that the project stops violating the GPL."

**Reply by 12people** [reply]
@Aga-C How is it not true? The FAQ question that you linked to specifically states:

> Note that there may also be legal issues with combining certain nonfree libraries with GPL-covered free software. Please see [the question on GPL software with GPL-incompatible libraries](https://www.gnu.org/licenses/gpl-faq.html.en#GPLIncompatibleLibs) for more information.

The rest of the answer is not related to legal matters.

**Reply by Aga-C** [reply]
@12people It doesn't mean that you can't use non-free libraries. It only states that you have to add a permission to the license what can be only done by copyright holder of the code and that you need to have copyright over all GPL licensed parts to do this (fortunately all external libs are MIT or Apache). Tibor hasn't done it, but it shouldn't be the problem if everyone agrees. Question is if we even need to check if everyone agrees when we just have this code already, and we can assume that Tibor did it after agreement with all copyright holders but just forgot to add proper term to the license.

**Reply by Aga-C** [reply]
Another thing from my original link:
> If the program is already written using the nonfree library, perhaps it is too late to change the decision. You may as well release the program as it stands, rather than not release it. But please mention in the README that the need for the nonfree library is a drawback, and suggest the task of changing the program so that it does the same job without the nonfree library. Please suggest that anyone who thinks of doing substantial further work on the program first free it from dependence on the nonfree library.

It doesn't say "you mustn't release". Also, another quote:
>suggest the task of changing the program so that it does the same job without the nonfree library

So, instead of a task `remove` we should have a task `rewrite simple editor to work the same as the advanced editor`.

**Reply by 12people** [reply]
@Aga-C The copyright holders in this case are all the historical contributors who have contributed to the code. This would include Tibor. You can't just assume that Tibor would agree to this GPL exception, you would have to get his express agreement. Which you can try to get, but I find it somewhat unlikely that he would want to help out a competing project.

(BTW, Tibor has been acting as if he owned the rights to all this code, selling it as a proprietary project to another company. He might be violating some laws here himself, but if he was the sole copyright holder, he would have been free to have his project under a dual license — proprietary with nonfree libs and then GPL for the code outside these libs.)

**Reply by 12people** [reply]
> It doesn't say "you mustn't release".

No, but the recommendation here isn't legal, it's philosophical. The FAQ assumes that you're the sole copyright holder here (referring to the program as "your program") and recommends that you release your code with an exception for this non-free lib rather than not release it at all.

It warns you that the code will be "entirely off limits to the Free World". And it also clearly tells you to check whether there might be potential legal issues with using this library.

**Reply by Aga-C** [reply]
>This would include Tibor. You can't just assume that Tibor would agree to this GPL exception, you would have to get his express agreement.

He added it himself to the code, so in fact he did this exception, but haven't edited the license properly (taking look at his current actions we can clearly see that he never meant to do things accordingly to GPL). However, that's such a strange, non-typical case that probably some lawyer who has expertise in GPL should take a stance here. If we consider not editing the license by Tibor as violating GPL for years, is it the only option now to remove this proprietary editor?

**Reply by 12people** [reply]
@Aga-C Tibor added it, but he was acting like the code was strictly his own, so he can be considered to be adding it to his own proprietary project.

Now, it seems like he might have been violating the GPL because he really wasn't the sole code contributor, but that doesn't automatically give us the rights to violate the GPL too.

**Reply by 12people** [reply]
(I'm assuming Tibor didn't have contributors sign a CLA — if he did, then he really might have been the sole copyright holder and was well within his rights to do anything he wanted with the project.)

**Reply by 12people** [reply]
One more note about this: it's actually not that uncommon for some projects to have a free software basis and then have a proprietary copy of that project with non-free additions on top of it.

This is basically the definition of an "open core" model. Gitlab is like this, for example. Or IntelliJ Studio.

There's also dual licensing, which is what e.g. Qt does — offer up free GPL and paid proprietary versions of the same software.

These projects have full ownership of the code, so they can do this without violating the GPL. If anyone else tried to do the same thing with their code, though, they would be violating the GPL.

**Reply by Aga-C** [reply]
@12people I've found the topic in Gallery about it (https://github.com/SimpleMobileTools/Simple-Gallery/issues/1706), and wanted to show that Tibor was in fact ignoring the license, but I've noticed that you reacted on his comment there 😄. I thought it was done by him in a more "normal" manner, not that he was just ignoring every legal fact from the start... I understand now the point why it should be removed after I've dug a bit more in older topics and that LICENSE is not modified in any way.

Let's leave this to @naveensingh to decide what he wants to do - refactor the code and remove the editor before the first release, or leave it like it is. For sure, it has to be done before license is bought.

**Reply by dp35654** [reply]
Hopefully you'll let those of us who completely use all these app's to run are phones! It would be very upsetting to wake up one day and have my phone default back too Google whom I haven't used since 2021, actually I didn't even have a smartphone till the sundowning of the 3G non Android phone system! So yes I would be furious after I have made sure Google didn't have access to my contacts

**Comment by inson1** [comment]
@Aga-C it was in discussions + naveensingh answered it, so that is why I didnt do any research  :) I changed the name + it needs to by removed anyway, because there arent going to be any pro versions and the sdk costs money - easiest would be to remove it :) - even if there would be pro versions I would be for removing it anyway
May you reopen it?

https://github.com/orgs/FossifyOrg/discussions/9#discussioncomment-7756115
Plus https://github.com/orgs/FossifyOrg/discussions/9#discussioncomment-7793145

**Comment by inson1** [comment]
@naveensingh may we add this to issues?

**Comment by naveensingh** [comment]
I haven't done any research on this yet so let's keep it here for now.

**Comment by pastk** [comment]
The Gallery Pro version distributed via F-Droid doesn't have this proprietary SDK, right?
I assume it uses some other (developed in-house?) editor, so there should be no need to refactor something for GP version, just use the F-Droid code version for GP releases too, is that right?

**Reply by Aga-C** [reply]
Code is the same, just paid editor is not bundled into F-Droids APK thanks to the build variant.

**Comment by naveensingh** [comment]
Ok, @12people thanks for pointing this out, the proprietary editor has been removed.

**Related:**

For the same reasons, any in-app-purchase donation option (using Google Play) can't be added to the project because the billing client library that is used to request purchases is closed-sourced. See https://issuetracker.google.com/issues/159151141. I'm not 100% percent sure and Google Play policies may have changed but last I checked direct donation links weren't allowed either so there's that (unless you are a registered charitable organization).

**Reply by inson1** [reply]
@naveensingh would non profit count as charitable organization?

**Reply by naveensingh** [reply]
@inson1 

A charitable organization is a type of non-profit organization, but not all non-profits are necessarily charitable organizations. It looks like they have replaced the 'charitable organization' part with 'Google Play's billing system must not be used in cases of, among others, tax-exempt donations'. An old outdated answer is [here](https://stackoverflow.com/a/35371309).

The payment policy received quite a few updates since the last time I checked:
https://support.google.com/googleplay/android-developer/answer/9858738

Unless I'm foggy and need more sleep, section 4 on that page seems to say we can't put donation links in the app. I have contacted the google play support team about this, hopefully they will provide some direct answer regarding this. If we still can't put direct donation links in the apps, pro versions are an alternative. We'll see, it's not a priority right now. The release is.

**Reply by naveensingh** [reply]
Answer from the Google Play developer support:

> Hi Naveen,
> 
> Thank you for contacting Google Play Developer Support Team.
> Please understand that while we do not currently offer a donation feature, if you're a certified 501(c) you may use Play in-app billing or a secure third party system to collect payments. Please note there is a 30% transaction fee for all Play in-app billing transactions.
> 
> If you are not 501(c) certified, you may not accept donations on an app provided through the Play Store.
> 
> If you have any further questions regarding Google Play policy, please visit the [Developer Policy Center](https://play.google.com/about/developer-content-policy.html) or contact your legal representative.

**Reply by inson1** [reply]
@naveensingh for example signal is 501(c)(3) 
idk if non profit from India would count. It doesnt seem like it.

**Reply by naveensingh** [reply]
I guess pro versions will do for now, with or without extra features (if the policy allows it).

**Reply by 12people** [reply]
As long as the extra features are GPL-licensed, it's fine. :)
